src/sentry_utils.c

```diff
@@ -376,3 +376,47 @@ sentry__msec_time_to_iso8601(uint64_t time)
     strcat(buf, "Z");
     return sentry__string_clone(buf);
 }
+
+uint64_t
+sentry__iso8601_to_msec(const char *iso)
+{
+    int len = strlen(iso);
+    // The code is adapted from: https://stackoverflow.com/a/26896792
+    int y, M, d, h, m, s, msec = 0;
+    int consumed = 0;
+    if (sscanf(iso, "%d-%d-%dT%d:%d:%d%n", &y, &M, &d, &h, &m, &s, &consumed)
+        < 6) {
+        return 0;
+    }
+    // we must have a tail, either `Z` or starting with `.`
+    if (consumed >= len || (iso[consumed] != 'Z' && iso[consumed] != '.')) {
```

Comment:
_nit_: Can you move the check for `Z` below consuming the decimal part, and remove it from the `sscanf` below?